client/ayon_core/hosts/nuke/plugins/publish/collect_headless_farm.py

```diff
@@ -0,0 +1,47 @@
+import pyblish.api
+
+
+class CollectHeadlessFarm(pyblish.api.ContextPlugin):
+    """Setup instances for headless farm submission."""
+
+    # Needs to be after CollectFromCreateContext
+    order = pyblish.api.CollectorOrder - 0.4
+    label = "Collect Headless Farm"
+    hosts = ["nuke"]
+
+    def process(self, context):
+        if not context.data.get("headless_farm", False):
+            return
+
+        for instance in context:
+            if instance.data["family"] == "workfile":
+                instance.data["active"] = False
+                continue
+
+            # Filter out all other instances.
+            node = instance.data["transientData"]["node"]
+            if node.name() != instance.context.data["node_name"]:
+                instance.data["active"] = False
+                continue
+
+            instance.data["families"].append("headless_farm")
+
+
+class SetupHeadlessFarm(pyblish.api.InstancePlugin):
+    """Setup instance for headless farm submission."""
+
+    order = pyblish.api.CollectorOrder + 0.4999
+    label = "Setup Headless Farm"
+    hosts = ["nuke"]
+    families = ["headless_farm"]
+
+    def process(self, instance):
+        # Enable for farm publishing.
+        instance.data["farm"] = True
+
+        # Clear the families as we only want the main family, ei. no review
+        # etc.
+        instance.data["families"] = ["headless_farm"]
+
+        # Use the workfile instead of published.
+        instance.data["use_published_workfile"] = False
```

Comment:
Related to comment https://github.com/ynput/ayon-core/pull/314/files#r1609616503
```suggestion
        attribute_values = instance.data.setdefault("attributeValues", {})
        attribute_values["use_published_workfile"] = False
```

Reply:
Looks like we need to use `publish_attributes` since `attributeValues` comes from `self.get_attr_values_from_data`; https://github.com/ynput/ayon-core/blob/develop/client/ayon_core/modules/deadline/plugins/publish/submit_nuke_deadline.py#L95